src/Components/Blazor/Build/src/targets/StaticWebAssets.targets

```diff
@@ -1,37 +1,65 @@
 <Project>
 
   <PropertyGroup>
-    <ResolveStaticWebAssetsInputsDependsOn>
-      $(ResolveStaticWebAssetsInputsDependsOn);
-      _RemoveBlazorCurrentProjectAssetsFromStaticWebAssets;
-    </ResolveStaticWebAssetsInputsDependsOn>
-
-    <GetCurrentProjectStaticWebAssetsDependsOn>
-      $(GetCurrentProjectStaticWebAssetsDependsOn);
-      _RemoveBlazorCurrentProjectAssetsFromStaticWebAssets;
-    </GetCurrentProjectStaticWebAssetsDependsOn>
+    <ResolveCurrentProjectStaticWebAssetsDependsOn>
+      _ResolveBlazorGeneratedAssets;
+      $(ResolveCurrentProjectStaticWebAssetsDependsOn);
+    </ResolveCurrentProjectStaticWebAssetsDependsOn>
   </PropertyGroup>
 
-
-  <Target Name="_RemoveBlazorCurrentProjectAssetsFromStaticWebAssets">
+  <Target Name="_ResolveBlazorGeneratedAssets" DependsOnTargets="PrepareBlazorOutputs">
     <ItemGroup>
-      <StaticWebAsset Remove="@(StaticWebAsset)" Condition="'%(SourceType)' == ''" />
+      <StaticWebAsset Include="@(BlazorOutputWithTargetPath->'$(TargetDir)%(TargetOutputPath)')" RemoveMetadata="TargetOutputPath">
+        <SourceType></SourceType>
+        <SourceId>$(PackageId)</SourceId>
+        <ContentRoot>$([MSBuild]::NormalizeDirectory('$(TargetDir)\dist\'))</ContentRoot>
+        <BasePath>$(StaticWebAssetBasePath)</BasePath>
+        <RelativePath>$([System.String]::Copy('%(TargetOutputPath)').Replace('dist\','').Replace('\','/'))</RelativePath>
+      </StaticWebAsset>
+
+      <!-- We are dependingo on a "private" property for static web assets, but this is something we can clean-up in a later release.
+           These files are not "external" in the "traditional" sense but it is fine for now as this is an implementation detail. -->
+      <_ExternalStaticWebAsset Include="@(BlazorOutputWithTargetPath->'$(TargetDir)%(TargetOutputPath)')">
+        <SourceId>$(PackageId)</SourceId>
+        <!-- We just do this to keep the existing implementation happy. We will update this in the next release. -->
+        <SourceType>Generated</SourceType>
+        <ContentRoot>$([MSBuild]::NormalizeDirectory('$(TargetDir)\dist\'))</ContentRoot>
+        <BasePath>$(StaticWebAssetBasePath)</BasePath>
+      </_ExternalStaticWebAsset>
     </ItemGroup>
+
   </Target>
 
   <Target Name="BlazorStaticWebAssetsComputeFilesToPublish"
     AfterTargets="_StaticWebAssetsComputeFilesToPublish">
 
     <ItemGroup>
       <!-- We need to update the external static web assets to follow the blazor publish output convention that puts them inside $(TargetName)/dist instead of wwwroot -->
-      <_StandaloneExternalPublishStaticWebAsset Include="@(_ExternalPublishStaticWebAsset)" Condition="'%(RelativePath)' != ''">
+      <_StandaloneExternalPublishStaticWebAsset Include="@(_ExternalPublishStaticWebAsset)" Condition="'%(RelativePath)' != '' and '$([System.String]::Copy(%(Identity))).StartsWith($(MSBuildThisProjectDirectory))' != 'true'">
```

Comment:
Where's `_ExternalPublishStaticWebAsset` defined?

Reply:
This is defined inside the Razor SDK on the _StaticWebAssetsComputeFilesToPublish target. I'm reusing the itemgroup here to avoid computing it again

Reply:
Not needed anymore